src/views/virtualmachines/actions/BulkVirtualMachineActionFactory.tsx

```diff
@@ -0,0 +1,166 @@
+import React from 'react';
+
+import VirtualMachineModel from '@kubevirt-ui/kubevirt-api/console/models/VirtualMachineModel';
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { LabelsModal } from '@kubevirt-utils/components/LabelsModal/LabelsModal';
+import { ModalComponent } from '@kubevirt-utils/components/ModalProvider/ModalProvider';
+import MoveBulkVMToFolderModal from '@kubevirt-utils/components/MoveVMToFolderModal/MoveBulkVMsToFolderModal';
+import { t } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { getLabels } from '@kubevirt-utils/resources/shared';
+import { Action, k8sPatch } from '@openshift-console/dynamic-plugin-sdk';
+import { VM_FOLDER_LABEL } from '@virtualmachines/tree/utils/constants';
+
+import ConfirmMultipleVMActionsModal from './components/ConfirmMultipleVMActionsModal/ConfirmMultipleVMActionsModal';
+import { ACTIONS_ID } from './hooks/constants';
+import { deleteVM, pauseVM, restartVM, startVM, stopVM, unpauseVM } from './actions';
+import { getCommonLabels, getLabelsDiffPatch } from './utils';
+
+export const BulkVirtualMachineActionFactory = {
+  delete: (vms: V1VirtualMachine[], createModal: (modal: ModalComponent) => void): Action => ({
+    cta: () =>
+      createModal(({ isOpen, onClose }) => (
+        <ConfirmMultipleVMActionsModal
+          action={deleteVM}
+          actionType="Delete"
+          isOpen={isOpen}
+          onClose={onClose}
+          vms={vms}
+        />
+      )),
+    id: ACTIONS_ID.DELETE,
+    label: t('Delete'),
+  }),
+  editLabel: (vms: V1VirtualMachine[], createModal: (modal: ModalComponent) => void): Action => ({
+    cta: () => {
+      const commonLabels = getCommonLabels(vms);
+
+      createModal(({ isOpen, onClose }) => (
+        <LabelsModal
+          onLabelsSubmit={(labels) => {
+            return Promise.all(
+              vms.map((vm) =>
+                k8sPatch<V1VirtualMachine>({
+                  data: getLabelsDiffPatch(labels, getLabels(vm)),
+                  model: VirtualMachineModel,
+                  resource: vm,
+                }),
+              ),
+            );
+          }}
+          initialLabels={commonLabels}
+          isOpen={isOpen}
+          obj={vms[0]}
```

Comment:
its guranteed? or need ?.

Reply:
I can add it anyway. Should be guaranteed